src/component/mxgraph/MxGraphRenderer.ts

```diff
@@ -163,16 +163,20 @@ function toDisplayedModel(bpmnModel: BpmnModel): DisplayedModel {
     })
     .map(shape => shape.bpmnElement?.id);
 
-  const subprocesses = bpmnModel.flowNodes.filter(shape => ShapeUtil.isSubProcess(shape.bpmnElement?.kind));
-
-  const boundaryEvents = bpmnModel.flowNodes.filter(shape => {
-    const kind = shape.bpmnElement?.kind;
-    return ShapeUtil.isBoundaryEvent(kind) && !ShapeUtil.isSubProcess(kind) && !collapsedSubProcessIds.includes(shape.bpmnElement?.parentId);
-  });
-
-  const otherFlowNodes = bpmnModel.flowNodes.filter(shape => {
+  const subprocesses: Shape[] = [];
+  const boundaryEvents: Shape[] = [];
+  const otherFlowNodes: Shape[] = [];
+  bpmnModel.flowNodes.forEach(shape => {
```

Comment:
👍 

ℹ️ If we change BPMNModel, no need this

Reply:
So what do we do for this code. Do we revert the last commit to restore the 3 array filters or do we keep it like this?
cc @aibcmars

Reply:
it's ok for me :)